package_metrics/parsing_utils.py

```diff
@@ -1,13 +1,32 @@
+import re
+
+
 def behind(latest, current):
     delta = [0, 0, 0]
-    for index, (lat, cur) in enumerate(zip(vsplit(latest), vsplit(current))):
+    parsed_latest = parse_version(latest)
+    parsed_current = parse_version(current)
+    for index, (lat, cur) in enumerate(zip(parsed_latest, parsed_current)):
         if lat != cur:
             delta[index] = lat - cur
             break
     return delta
 
 
-def vsplit(version):
+def parse_version(version):
+    """
+    :param version: a string representing the package version (e.g., '1.0.1')
+    :return: a list of integers representing [major, minor, patch]
+    """
     padded = f"{version}.0.0"  # append extra zeros for versions like '1' or '2.0'
-    return [int(n) for n in padded.split(".")[:3]]
+    semvar_version = []
+    for n in padded.split(".")[:3]:
+        try:
+            num = int(n)
+        except ValueError:
+            match = re.search(r'\d+', n)
```

Comment:
This will find the first number anywhere in the string. I think it should be either `re.match(r'\d+', n)` or `re.search(r'^\d+', n)` - both are equivalent.

Suggested additional test: `"3.a3.1"` -> `[3, 0, 1]`

Reply:
Fixed in 903349ad7484158bc3ce4f95c32c8b0c368c2da5